File: algorithmns/ArrayAndSort/QuickSort1.java
import java.util.*;
public class QuickSort1 {

	static void partition(int[] ar) {
		int N = ar.length;
		int pivot = ar[0];
		ArrayList<Integer> left = new ArrayList<Integer>();
		ArrayList<Integer> right = new ArrayList<Integer>();

		for(int i = 1; i<N; i++){
			if(ar[i] <= pivot) left.add(ar[i]);
			else right.add(ar[i]);
		}
		for (int k=0; k<left.size(); k++ )
			ar[k] = left.get(k);
		ar[left.size()] = pivot;
		
		for (int l=0; l<right.size(); l++ )
			ar[left.size()+1+l] = right.get(l);
		printArray(ar);
	}
 
	static void printArray(int[] ar) {
		for(int n: ar){
			System.out.print(n+" ");
		}
		 System.out.println("");
	}
			 
	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
		int n = in.nextInt();
		int[] ar = new int[n];
		for(int i=0;i<n;i++){
			ar[i]=in.nextInt(); 
		}
		partition(ar);
	}
}

File: algorithmns/ArrayAndSort/QuickSort1.rb
def  partition( ar) 
	n = ar.count;
	pivot = ar.shift
	left = Array.new()
	right = Array.new()

	ar.each do |value|
		if value <= pivot
			left.push(value)
		else
			right.push(value)
		end
	end
	for i in 0..left.count
		ar[i] = left[i]
	end
	ar[left.count] = pivot
	for i in 0..right.count
		ar[i+1+left.count] = right[i]
	end
	ar.each { |a| print a, " " }
end

cnt = gets.to_i;
ar = STDIN.gets.chomp.split(" ");
partition(ar);
